Dshop/apps/users/serializers.py

```diff
@@ -0,0 +1,43 @@
+from rest_framework import serializers
+from rest_framework.serializers import ValidationError
+from django.utils.translation import gettext_lazy as _
+from django.contrib.auth import get_user_model
+
+User = get_user_model()
+
+
+class RegistrationSerializer(serializers.Serializer):
+    """
+    Serializer related to registration of a new user
+    """
+    username = serializers.CharField(min_length=2, max_length=255)
+    email = serializers.EmailField(max_length=255)
+    password1 = serializers.CharField(write_only=True)
+    password2 = serializers.CharField(write_only=True)
+
+    def validate_username(self, username):
+        if User.objects.filter(username=username).exists():
```

Comment:
Username is unique, if you use modelSerializer it will be worikng automatically,

Reply:
Or use https://www.django-rest-framework.org/api-guide/validators/#uniquevalidator